run.py

```
@@ -227,24 +290,121 @@ async def get_historic_payments() -> List[Dict[str, Any]]:
+        except FileNotFoundError:
+            logger.warning("Collateral data file not found")
+            return []
+
+        if collateral_df is None or collateral_df.empty:
+            return []
+
+        return collateral_df.to_dict("records")
+
+    except HTTPException:
+        raise
+    except Exception as e:
```

Ruff (0.13.3)

Do not catch blind exception: `Exception`

(BLE001)